server/models/OAuthAuthorizationCode.ts

```diff
@@ -84,6 +85,20 @@ OAuthAuthorizationCode.init(
     sequelize,
     tableName: 'OAuthAuthorizationCodes',
     paranoid: true, // For soft-deletion
+    hooks: {
+      async afterCreate(oauthAuthorizationCode: OAuthAuthorizationCode) {
+        const application = await models.Application.findByPk(oauthAuthorizationCode.ApplicationId);
+
+        models.Activity.create({
+          type: activities.OAUTH_APPLICATION_AUTHORIZED,
+          UserId: oauthAuthorizationCode.UserId,
+          data: {
+            application: application.publicInfo,
+            scope: oauthAuthorizationCode.scope || [],
+          },
+        });
+      },
+    },
```

Comment:
This is a clever solution, and technically this should work.
At Open Collective we tend to avoid unexplicit side-effects since they rely too much on context and can make debugging difficult. It is also good to write explicit (and sometimes dumb) code because that helps people outside our core-team to contribute.

So, for consistency, I would rather see this activity being created in `lib/oauth/model.ts` inside the `saveAuthorizationCode` function.
If you're up for another challenge, it would also be nice to update that function unit test to check if the activity was created.

Reply:
Thanks @kewitz! I made this change.